src/drawers/canvasDrawer.ts

```diff
@@ -20,6 +20,9 @@ export type CanvasDrawStep = (
  * This class is immutable (but has internal state) and shouldn't be extended.
  */
 export class CanvasDrawer implements IDrawer {
+  protected _lastDrawnData: any[];
+  protected _lastProjector: AttributeToAppliedProjector;
```

Comment:
Would avoid adding state here -- stateful drawers complicate thing (such as how we now need a drawerFactory in ProxyDrawer). This state belongs in the Plot itself I think.